### ISSUE TITLE
👩‍💻📞 WinUI Community Call (May 20, 2020)

### ISSUE THREAD

**Issue by anawishnoff** [microsoft/microsoft-ui-xaml#2437]
This community call live stream has ended, but you can watch the recording here:

https://www.youtube.com/watch?v=TeS4rX3i_Ls

## Details
Date: Wednesday May 20, 2020
Time: 16:00-17:00 UTC (9:00-10:00am Pacific)

Anyone and everyone is welcome - no pre-registration is required.
This will be an informal interactive live stream directly with members of our engineering team.

## Format
In this community call we'll be doing a deeper dive into WinUI 3 Preview 1 and sharing some bonus content from our team's //Build sessions, alongside some of our partners.

At the end of every community call we have a Q&A session where our team answers your questions about all things WinUI. **Ask your questions in the comments of this issue**!

## Agenda
•	Intro/agenda 
•	WinUI 3 Preview 1 capabilities, limitations, and roadmap updates w/ @ryandemopoulos 
•	Quick mention of Windows Community Toolkit updates w/ @michael-hawker 
•	Deep-dive: Preview 1 demo w/ @marb2000 
•	Partner Spotlight: @dotMorten of Esri
•	Partner Spotlight: Uno Platform 
•	Quick Q&A

**Comment by chingucoding** [comment]
A great question from @mdtauk : #2439
Will the Segoe MDL2 icons be replaced with the fluent icon system?

Something I wonder: Once WinUI 3 has been released, are there already plans on what will done next? Any controls that are planned, that will be developed (immediately) once WinUI 3 shipped?

**Comment by mdtauk** [comment]
@chingucoding  I know Input Validation is on that todo list.

Also there is the DataGrid and WinUI Ribbon.  I guess there is stuff from the freezer to be considered.

**Comment by Felix-Dev** [comment]
@mdtauk Input Validation will come with WinUI 3.0 if nothing unexpected happens.

**Comment by carmineos** [comment]
Any info about https://github.com/microsoft/microsoft-ui-xaml/issues/668 and https://github.com/microsoft/microsoft-ui-xaml/issues/1080 would be appreciated

**Comment by mdtauk** [comment]
> @mdtauk Input Validation will come with WinUI 3.0 if nothing unexpected happens.
>
I thought it had been scaled back quite a bit to hit 3.0 release?

As for other controls, some I have noted are:
- Long Lived Status Messages; #913 
- Pagination Control; #60 
- UWP Ribbon; #168 

Some proposals I would like to see personally:
- Task Dialog; #2313 
- Prefix and Suffix for TextBoxes; #784 
- Pivot fit and finish; #940 #844 
- Updated Light and Dark theme colours; #920 
- Better Shadow support; #1009 
- Updates to form controls for Density with flyouts;
- Updates to various form controls for better keyboarding; #918 #2255 

In general I think we need to go over most UI to align where possible with Fluent Web and other post Windows 10 UWP UI frameworks.  WinUI is lacking in some areas, some controls missing.  

Windows 10X could be a catalyst for hurrying the pace of updating the UI and controls - hopefully inbox apps that have remained untouched can get updated too.  Of course that would be helped by Microsoft allowing Insiders to install versions on devices (if anyone is listening, I would be happy to run it on my Surface Go 🙂).  So feedback can be submitted.

**Comment by Felix-Dev** [comment]
@mdtauk To my understanding it wasn't scaled back. Instead, some folks have certain extra wishes (like support for attributes such as `MaxLength(n)`) which would likely only arrive post WinUI 3.0. Input Validation will work just fine with WinUI 3.0 (i..e checking for a min/max input length).

Working on new controls immediately after WinUI 3.0 is out is nice and all but in my view, signinifcant resources should be invested in addressing the many bugs and small feature requests which accumulated to hundreds of issues here in the WinUI repo. 

We also have a few controls which have been in preview for months now, such as the new ScrollViewer control. Those should be finished first before resources are diverted to work on entirely new controls.

**Comment by chingucoding** [comment]
I think to summarize/reiterate over the last few comments: What are the immediate plans after WinUI 3 release? What are the next steps steps the team is planning to do after the WinUI 3 release?

**Comment by robloo** [comment]
> Working on new controls immediately after WinUI 3.0 is out is nice and all but in my view, significant resources should be invested in addressing the many bugs and small feature requests which accumulated to hundreds of issues here in the WinUI repo. In a priority-based approach, of course.
>
>We also have a few controls which have been in preview for months now, such as the new ScrollViewer control. Those should be finished first before resources are diverted to work on entirely new controls.

Microsoft **must** spend the time after the initial WinUI 3.0 release to fix the open bugs. This is the best way to get confidence back in Microsoft XAML tooling. It will also make development a joy again which will spur new ideas, new applications and new developers. I continuously find bugs that would never happen in the WPF days.

**Comment by gopal-amlekar** [comment]
Hi, a very basic question. Sorry if that is already answered somewhere else. Apps developed with WinUI with .net core - Will they run on Windows 7 as well?

Thanks a lot,

**Comment by mdtauk** [comment]
> Hi, a very basic question. Sorry if that is already answered somewhere else. Apps developed with WinUI with .net core - Will they run on Windows 7 as well?
> 
> Thanks a lot,
>
I think it is safe to assume WinUI still depends on Windows 10, and for Windows 7 or 8 you will need to use WPF, or WinForms - which are also now Open Source

**Comment by chingucoding** [comment]
Another option you can use is the Uno project: https://platform.uno/

That allows you to run WinUI apps on Windows 7, the web and mobile platforms.

**Comment by gopal-amlekar** [comment]
> I think it is safe to assume WinUI still depends on Windows 10, and for Windows 7 or 8 you will need to use WPF, or WinForms - which are also now Open Source

Yes, we are using WPF. (Can XAML islands be used for an app targeting compatibility with Windows 7?)

> Another option you can use is the Uno project: https://platform.uno/
> That allows you to run WinUI apps on Windows 7, the web and mobile platforms.
That is great, yes I am tracking Uno as well. 

Thank you very much for your replies @chingucoding !

**Comment by ShankarBUS** [comment]
Hey @gopal-amlekar, how about using [Kinnara/ModernWpf](https://github.com/Kinnara/ModernWpf). It has necessary controls and styles for WPF to match WinUI and will be useful for non-win10 projects

**Comment by weitzhandler** [comment]
Please talk about input validation. It's a deal breaker and I'm so glad it's in progress. Please elaborate on your implementation plans.
Thanks folks, you're awesome!

**Comment by massijay** [comment]
News about ink controls like InkCanvas, InkToolbar? It would be also nice to have ink integration in the new Webview2 (like the old Edge)

**Comment by smourier** [comment]
I'm very excited with the WinUI plan (at last a serious UI strategy for Windows), and I can't wait for it :-) My questions:

I understand WinUI will run only on Windows 10 (as I understand, uno on Windows 7 is "just" PWA or electron based) , but what version explicitly as there are many? Will it be possible to use some features on 1703, others 1809, and so on?

Also any reasonable WinUI 3 release date?

Also, what will be the dependencies (like dll or nugets)? For example, if I write a "pure" .NET Framework or .NET Core app using C# (not UWP, not winrt, not c++), will I need Win2D with its nuts-driving MSVC deps deployment issues? This is just an example.

Also, will I be able to use the raw DirectX or Direct2D surfaces/rendertarget/devicecontexts below? without Win2D?

Will Xaml Islands will go? (as show in the roadmap picture?) Or is just they will become useless?

Also, what's the relation between Windows.UI.Xaml and Microsoft.UI.Xaml? why these two?

**Comment by Felix-Dev** [comment]
@smourier 
> I understand WinUI will run only on Windows 10 (as I understand, uno on Windows 7 is "just" PWA or electron based) , but what version explicitly as there are many? Will it be possible to use some features on 1703, others 1809, and so on?

WinUI 3 will run on 1803 Windows 10 and above. If you need to target even older Windows 10 versions, you will need to use WinUI 2 (at least 1703 and above).

> Also any reasonable WinUI 3 release date?
A first WinUI preview will be released around Build next week. Release of WinUI 3 is currently slated for end of the year.

> Will Xaml Islands will go? (as show in the roadmap picture?) Or is just they will become useless?
XAML Islands will be *included* in WinUI 3 going forward. You can still modify your existing Win32 apps progressively with WinUI 3 like you can with XAML Islands now.

> Also, what's the relation between Windows.UI.Xaml and Microsoft.UI.Xaml? why these two?
Windows.UI.Xaml.* is the namespace for the OS UWP UI APIs. These won't go away (as they ship with the OS). No investments will happen in them any longer though, only necessary (security) support if required. You can still write an UWP app which will not use WinUI. Such an app will use that namespace.

Microsoft.UI.Xaml is the namespace of the WinUI components (controls, input APIs,...). If you are using WinUI 2 already you are already using *both* namespaces. Some controls like NavigationView exist both in the OS and WinUI 2, the namespace will indicate which version you are using in your app.

On a sidenote, you won't be able to mix both Windows.UI.Xaml.* and Microsoft.UI.Xaml.* in a WinUI 3 app.

**Comment by jtorjo** [comment]
Compilation times please - will these be improved any time soon? Or do we need to wait another year?
(my app right now compiles in about 2 minutes 13 seconds, which is insanely much)

**Comment by cirrusone** [comment]
I have a few questions
1) What timescales/features are likely for new datagrid?
2) Will community toolkit datagrid work for potential preview release in May on Win32?
3) Is there a propertygrid in the pipeline for WinUI?
4) Is there a roadmap outlining what is likely to be delivered and when?

**Comment by michael-hawker** [comment]
🦙 @cirrusone some quick initial info for you:

1. I'm assuming they won't work on new DataGrid until after WinUI 3 ships, so see 4 below.
2. We've been doing some tests of the Toolkit with WinUI 3, but not sure if we've tested the current DataGrid yet - I'll have to check. However, to use the Toolkit with WinUI 3 you would need a version of the Toolkit built for WinUI 3. While we have a branch we're working from, it's not trivial to build due to dependencies which also need to support WinUI 3. I'll have an update on this work on the call though, so tune in and join us!
3. PropertyGrid is an interesting and complex control. We kind of have a modern internal one for the Toolkit's sample app, but it's not generalized enough to ship. If you want to [open an issue on the toolkit repo](https://github.com/windows-toolkit/WindowsCommunityToolkit/issues) we can certainly start a discussion around what it'd take to lift ours as a starting point and going from there.
4. The [roadmap](https://github.com/microsoft/microsoft-ui-xaml/blob/master/docs/roadmap.md) for WinUI is here, though @ryandemopoulos maybe it could get an update based on all the info mentioned in community calls this year? In the meantime check out the recording of the last couple of community calls as they've shared updates there.

**Comment by zhuxb711** [comment]
Even after a long period of investigation, WinUI Team cannot found the cause of the following problems. I hope these very obvious bugs can be resolved as soon as possible.
#1820 #2040 #2331

**Comment by Pinox** [comment]
Is there any progress / timeframe on nugets for us to convert existing UWP to  UWP (winui3.0) + .net core.

When are we going to see :
1) Winui3.0  Preview Nugets
2) Winui3.0 templates for UWP/.net core
3) Windows Community Preview (based on winui3.0) nugets.

Too really add value / spot issues we need the above to start converting existing projects.

**Comment by msavazzi** [comment]
Is it possible to have an update and a discussion of all the bugs I'm finding on TreeView?

**Comment by dotMorten** [comment]
@pinox
> When are we going to see :
>   Winui3.0 Preview Nugets
>   Winui3.0 templates for UWP/.net core

Both those things were already in the alpha (except .net core support which they didn't commit to supporting until preview)

**Comment by MartinZikmund** [comment]
Yes! Excited to see Uno Platform on the schedule! 🎉

**Comment by leoniDEV** [comment]
I just finished seeing [The Journey to One .NET](https://mybuild.microsoft.com/sessions/b7e27509-c56c-42ad-9ce2-34270ecb0a38) session from [Build 2020](https://mybuild.microsoft.com/home) and heared about [Multi-platform App UI (MAUI)](https://github.com/dotnet/maui).
So the questions are what will be the relation (if there will be a relation) between this two projects?
There will be integration between WinUI 3 and MAUI?
We will be able to create WinUI3 apps through MAUI? There will be a convergence of the XAML dialects?

**Comment by lhak** [comment]
Will Microsoft provide a preview version of win2d for winui 3.0 soon?

**Comment by kakone** [comment]
> Will the Segoe MDL2 icons be replaced with the fluent icon system?

Very good question. And please update the license of the Segoe MDL2 icons font, we want to use it on other platforms than Windows with Uno Platform.

**Comment by wbokkers** [comment]
Some guidance on porting a UWP app with desktop extensions to WinUI for Desktop would be great. (activation, getting win32 companion app in the main UWP app, conversion tools)
